command/transform/transformation.go

```diff
@@ -56,7 +58,7 @@ func (t *Transformation) Run(parser *dockerfile.Parser) error {
 	}
 
 	file.From.Emit(t.Output)
-	return t.write(parser, &file, "")
+	return t.write(parser, &file, "", map[string]bool{file.From.Image: true})
```

Comment:
Something like this, but with methods that encapsulate what's happening in lines 81-84 and the check in line 118:

```
func  (p *Provides) add(images []string)
func (p *Provides) include(image) bool
```